presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -116,6 +119,9 @@
             .toFormatter()
             .withOffsetParsed();
 
+    // the default date when the Gregorian calendar was instituted (October 15, 1582)
+    private static final long GREGORIAN_CALENDAR_INTRODUCTION = new GregorianCalendar().getGregorianChange().getTime();
```

Comment:
What do you mean by "the default" here?

The cutoff date defaults to 1582-10-15, but you are not using a constant.
Instead youre asking for locale/env specific date. Maybe

```
// The date when the Gregorian calendar was instituted, environment specific. Defaults to October 15, 1582.
```

?

Reply:
> Anyway, after adding some more tests, it turns out that Joda millisecond values are not consistent with java.sql.Date even for more recent dates (as late as the 1800's), so I decided to use Joda only starting with the 20th century.

I forgot about that, but indeed. I think i remember finding a zone where _java time_ and _java.util.Calendar_ differed > 1900 (but before 1950).
_java time_ and _java.util.Calendar_ use the same tz data file, but they seem to parse it differently.

... I can only find as recent as 1890s in multiple zones (eg Europe/Warsaw or Asia/Aden), so maybe my memory is wrong. Using 1900 seems to be safe.

_java time_ and _joda_ seem to behave the same as long as they have the same tz data. But tz data shouldn't change for the past date/times.

This is some variation of the test code i am using: https://github.com/findepi/urandom-bits/blob/master/src/main/java/findepi/time/JodaJdkZoneDrift.java

Reply:
Nice, that's good info.